libcaf_test/caf/test/fixture/flow.hpp

```diff
@@ -203,6 +278,12 @@ class CAF_TEST_EXPORT flow {
     return coordinator()->add_child(std::in_place_type<auto_observer<T>>);
   }
 
+  /// Returns a new canceling observer.
+  template <class T>
+  auto make_canceling_observer(bool accept_first = false) {
```

Comment:
It's not super obvious what this will do, so I'd like to see some documentation on this. something like this:

```
  /// Returns a new canceling observer. The subscriber will either call `cancel` on
  /// its subscription immediately in `on_subscribe` or wait until the first
  /// call to `on_next` when setting `accept_first_subscription` to `true`.
  template <class T>
  auto make_canceling_observer(bool accept_first_subscription = false) {
```